clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -1236,6 +1316,12 @@ public void close() {
             this.metadata.close();
         }
 
+        private void rebootstrap(long now) {
+            closeAll();
+            metadata.rebootstrap();
+            metadataAttemptStartMs = Optional.of(now);
```

Comment:
Question: Do we need to set the time while rebootstrapping or reset to `Optional.empty`? As whenever there would be successful response from metadata then the time will be updated?

Or is it more to handle that after re-bootstrap, if we never receive successful metadata response then we would like to rebootstrap again?

Reply:
It is to handle rebootstrapping again if successful metadata response is not received after one rebootstrap (the second case you mentioned).